oauth2-server/src/main/java/com/clouway/oauth2/codechallenge/AuthorizationCodeVerifier.java

```diff
@@ -0,0 +1,48 @@
+package com.clouway.oauth2.codechallenge;
+
+import com.google.common.base.Strings;
+import com.google.common.io.BaseEncoding;
+
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+/**
+ * @author Vasil Mitov <vasil.mitov@clouway.com>
+ */
+public class AuthorizationCodeVerifier implements CodeVerifier {
+  @Override
+  public boolean verify(CodeChallenge codeChallenge, String codeVerifier) {
+    try {
+      //no code verifier was provided and no code challenge was saved so this is a normal OAuth2 code flow
+      if (Strings.isNullOrEmpty(codeVerifier) && isNullOrEmpty(codeChallenge)) {
```

Comment:
codeChallengeCannot be empty as it's not a string value. 

// code challenge was provided or not 
// code challenge was defined or not 
if (!codeChallenge.isProvided()) {
}

but you have to ensure that this instance is always created and null is not passed

Reply:
In the ResourceOwnerClientAuthorizationTest it is shown that if it is not provided an empty CodeChallenge is saved.